src/xrGamepad.js

```diff
@@ -0,0 +1,76 @@
+const Constants = require("./constants.js");
+const XRGamepadComponent = require("./xrGamepadComponent.js");
+
+/**
+  * @description This class connects an XR gamepad, as described in a 
+  * mapping.json file, with a Gamepad object, as defined 
+  * by https://www.w3.org/TR/gamepad/
+  * @author Nell Waliczek / https://github.com/NellWaliczek
+*/
+class XRGamepad {
+
+  /**
+   * @param {Object} gamepad - The Gamepad object provided by the user agent as 
+   * defined by https://www.w3.org/TR/gamepad/
+   * @param {Object} mapping - The mapping file text in dictionary form
+   * @param {String} handedness - An enum value as defined by Handedness in 
+   * constants.js
+   */
+  constructor(gamepad, mapping, handedness) {
+    if (!gamepad || !mapping) {
+      throw new Error("Cannot create XRGamepad from invalid parameters");
+    }
+
+    if (gamepad.id != mapping.id) {
+      throw new Error(`Gamepad id ${gamepad.id} and mapping id ${mapping.id} do not match`);
+    }
+
+    if (gamepad.handedness && gamepad.handedness != handedness) {
```

Comment:
What's the benefit to passing in the handedness here, as opposed to just picking up the one that's reported by the gamepad? This largely seems like it's just forcing developers to pass in gamepad.handedness again and slapping them on the wrist if they don't?

Reply:
WebXR won't be repording handedness as part of the gamepad, yes?  It's just WebVR that does that.  I could change the check to make handedness optional and only error out if it's not on the gamepad...wdyt?

Reply:
Ah, yes. I'd forgotten about the difference in WebVR/XR here. Don't mind me! This is fine.